src/communication/xcWebSocketPublisher.js

```diff
@@ -21,6 +21,17 @@ define(function () {
     }
 
 
+	Publisher.prototype.canPublish = function (componentName, stateMachineName, messageType) {
+	    try {
+	        var codes = this.configuration.getCodes(componentName, stateMachineName);
+	        this.configuration.getPublisherDetails(codes.componentCode, codes.stateMachineCode, messageType);
+	        return true;
+	    } catch (e) {
```

Comment:
It's not the best idea to rely on exception. It's better to really check if I can publish...